week9/answers.sql

```diff
@@ -1,3 +1,193 @@
 -- Create your tables, views, functions and procedures here!
 CREATE SCHEMA social;
 USE social;
+
+CREATE TABLE users (
+	user_id INT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT,
+	first_name VARCHAR(30) NOT NULL,
+	last_name VARCHAR(30) NOT NULL,
+	email TEXT NOT NULL,
+	-- Found current timestamp here:
+	-- https://www.w3schools.com/sql/func_sqlserver_current_timestamp.asp
+	created_on DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+
+CREATE TABLE sessions (
+  session_id INT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT,
+  user_id INT UNSIGNED NOT NULL,
+  created_on DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_on DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  CONSTRAINT sessions_fk_users
+    FOREIGN KEY (user_id)
+    REFERENCES users (user_id)
+      ON UPDATE CASCADE
+      ON DELETE CASCADE
+);
+
+DELIMITER ;;
+CREATE TRIGGER session_update_time
```

Comment:
Clever solution, but see my comment above to do all of this by adding three words in your table definition.